### ISSUE TITLE
Ragged Tensors

### ISSUE THREAD

**Issue by PatReis** [keras-team/keras#18414]
Hello,

thanks for the wonderful work on Keras-Core.
I saw the issue keras-team/keras-core#670 and keras-team/keras-core#223 and wanted to ask about the ideas/roadmap to support ragged tensors.
Will there be a possibilty to pass a `KerasRaggedTensor` to layers and models, of course without the support of ragged operations across backends ? To become compatible with `tf.keras` ? Just to feed a ragged shaped tensor to model or layers without having to resort to padding...
I know that this is probably a lot of work, so I just wanted to know in order to plan ahead for myself.

Best regards

**Comment by PatReis** [comment]
@swamidass 
So I am currently trying to port my old keras graph library (https://github.com/aimat-lab/gcnn_keras/tree/master) to keras 3.0. And I decided to use disjoint implementation of PyTorch Geometric as main graph representation. 
I think jraph and DGL use it too. 
This can be realized as there is not restriction on tensors between layers.
It seems to work to pass ragged tensors to keras 3.0 models already. Although a ragged kwarg in Input layer would be great for backward compatibilty.
So I ended up to simply decompose them in the first layer and continue with normal disjoint tensors (tested in tensorflow only yet).
For jax however you would have to use a loader either PyTorch data loader or tf.data to load disjoint with padded and fixed size into the model. I have not tested this yet. For Jax, bucketing and padding is the only way I think. But I believe that padded disjoint with a dummy graph at padded index would not yield a great performance reduction.

**Comment by swamidass** [comment]
For Jax, it is easiest because there are no constraints. For performance reasons, you do want to pad to consistent sizes, and Jraph has a simple function to accomplish this. Nonetheless, the input tensors have different sizes leading dimensions regardless.

@PatReis, how are you managing loading batches into keras fit? Or are you jut avoiding keras fit and writing your own training function?

**Comment by PatReis** [comment]
@swamidass 
No, I am still working on the port to keras 3 and everything is experimental at the moment. But here is an example on how you could realize loading with keras fit and future `kgcnn` package: 

https://github.com/aimat-lab/gcnn_keras/blob/master/docs/source/models.ipynb
or
https://github.com/aimat-lab/gcnn_keras/blob/master/notebooks/tutorial_model_loading_options.ipynb

**Comment by shkarupa-alex** [comment]
+1 to restore ragged tensors support

**Comment by swamidass** [comment]
What is the current status of keras and ragged tensors?

**Comment by hertschuh** [comment]
Some ragged support was added for the Tensorflow backend. In particular `KerasTensor(shape, ragged=True)`.

Resolving this bug. Let us know if a particular feature is missing.
